app/invocation/workflow_rerun_button.tsx

```diff
@@ -82,21 +91,33 @@ export default class WorkflowRerunButton extends React.Component<WorkflowRerunBu
       }
     }
 
+    const req = new workflow.ExecuteWorkflowRequest({
+      workflowId: configuredEvent.workflowId,
+      actionNames: [configuredEvent.actionName],
+      pushedRepoUrl: configuredEvent.pushedRepoUrl,
+      pushedBranch: configuredEvent.pushedBranch,
+      commitSha: configuredEvent.commitSha,
+      targetRepoUrl: configuredEvent.targetRepoUrl,
+      targetBranch: configuredEvent.targetBranch,
+      visibility: this.props.model.buildMetadataMap.get("VISIBILITY") || "",
+      pullRequestNumber: Long.fromString(this.props.model.buildMetadataMap.get("PULL_REQUEST_NUMBER") || "0"),
+      async: true,
+    });
+    try {
+      req.env = await this.getEnvVarsForWorkflow(workflowExecution!);
+    } catch (e) {
+      // If the workflow was executed by API, env vars could've been overwritten,
+      // so we need to fetch them from the workflow action.
+      // If the action has expired, return an error.
+      if (configuredEvent.actionTriggerEvent == "manual_dispatch") {
+        this.setState({ isLoading: false });
+        errorService.handleError(`Failed to rerun manually dispatched execution: ${e}.`);
+        return;
+      }
+    }
```

Comment:
Seems we should be more deterministic and either always try to set the `req.env` (and fail if we can't), or don't try to set `req.env` at all:

```suggestion
    if (configuredEvent.actionTriggerEvent == "manual_dispatch") {
      try {
        req.env = await this.getEnvVarsForWorkflow(workflowExecution!);
      } catch (e) {
        // If the workflow was executed by API, env vars could've been overwritten,
        // so we need to fetch them from the workflow action.
        // If the action has expired, return an error.
        this.setState({ isLoading: false });
        errorService.handleError(`Failed to rerun manually dispatched execution: ${e}.`);
        return;
      }
    }
```